src/request.js

```diff
@@ -0,0 +1,21 @@
+import superagent from 'superagent'
+import Config from './config'
+import Logger from './logger'
+
+export default class Request {
+  constructor(uri, mimeType) {
+    this.mimeType = mimeType || Config.defaultMimeType
+    this.logger = new Logger()
+    this.uri = uri
+    this.agent = superagent.get(this.uri).accept(this.mimeType)
```

Comment:
Funnily enough, I used to have this named `this.request` instead of `this.agent`. I renamed it to `this.agent` for two reasons:

1. Having a `Request` instance with an inner `this.request` attribute felt weird and redundant to me. (`this.agentRequest` is better in this regard.)
2. Maybe more of a concern: at this point, the request hasn't been made yet. All the pieces are in place, and the request fires off once `.then()` is called on the agent. So it's not yet a request *per se*.

I don't feel too strongly about this, though. How about you?

Reply:
ah, yeah, ok.  i can see it as an agent that way, since it's yet to act, and doing a thing on our behalf.  also don't feel strongly, happy to leave as is.